cmd/rpcdaemon/commands/trace_api.go

```diff
@@ -0,0 +1,246 @@
+
+					blockNumbers, err := retrieveHistory(tx, addr)
+					addrBytes := addr.Bytes()
+					if err != nil {
+						return err
+					}
+
+					for _, num := range blockNumbers {
+						if req.FromBlock != nil && uint64(*req.FromBlock) > num {
+							continue
+						}
+
+						if req.ToBlock != nil && uint64(*req.ToBlock) < num {
+							continue
+						}
+
+						block := rawdb.ReadBlockByNumber(api.dbReader, num)
+						txs := block.Transactions()
+						senders := rawdb.ReadSenders(api.dbReader, block.Hash(), num)
+						for i, tx := range txs {
+							if uint64(len(filteredTransactionsHash)) == maxTracesCount {
+								if uint64(len(filteredTransactionsHash)) == api.maxTraces {
+									return fmt.Errorf("too many traces found")
+								}
+								return nil
+							}
+
+							if !isAddressInFilter(&senders[i], req.FromAddress) {
+								continue
+							}
+							var to *common.Address
+							if tx.To() == nil {
+								to = &common.Address{}
+							} else {
+								to = tx.To()
+							}
+
+							if bytes.Equal(to.Bytes(), addrBytes) {
+								filteredTransactionsHash = append(filteredTransactionsHash, tx.Hash())
+							}
+						}
+					}
+				}
+			}
+		} else if req.FromBlock != nil || req.ToBlock != nil { // iterate over blocks
+			var fromBlock uint64
+			var toBlock uint64
+			if req.FromBlock == nil {
+				fromBlock = 0
+			} else {
+				fromBlock = uint64(*req.FromBlock)
+			}
+
+			if req.ToBlock == nil {
+				headNumber := rawdb.ReadHeaderNumber(api.dbReader, rawdb.ReadHeadHeaderHash(api.dbReader))
+				toBlock = *headNumber
+			} else {
+				toBlock = uint64(*req.ToBlock)
+			}
+			if fromBlock > toBlock {
+				return fmt.Errorf("invalid parameters: toBlock must be greater than fromBlock")
+			}
+
+			for blockNum := fromBlock; blockNum < toBlock+1; blockNum++ {
+				block := rawdb.ReadBlockByNumber(api.dbReader, blockNum)
+				blockTransactions := block.Transactions()
+				for _, tx := range blockTransactions {
+					if uint64(len(filteredTransactionsHash)) == maxTracesCount {
+						if uint64(len(filteredTransactionsHash)) == api.maxTraces {
+							return fmt.Errorf("too many traces found")
+						}
+						return nil
+					}
+					filteredTransactionsHash = append(filteredTransactionsHash, tx.Hash())
+				}
+			}
+		} else {
+			return fmt.Errorf("invalid parameters")
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	getter := adapter.NewBlockGetter(api.dbReader)
+	chainContext := adapter.NewChainContext(api.dbReader)
+	genesisHash := rawdb.ReadBlockByNumber(api.dbReader, 0).Hash()
+	chainConfig := rawdb.ReadChainConfig(api.dbReader, genesisHash)
+	for _, hash := range filteredTransactionsHash {
+		_, blockHash, _, txIndex := rawdb.ReadTransaction(api.dbReader, hash)
+		msg, vmctx, ibs, _, err := transactions.ComputeTxEnv(ctx, getter, chainConfig, chainContext, api.db, blockHash, txIndex)
+		if err != nil {
+			return nil, err
+		}
+		trace, err := transactions.TraceTx(ctx, msg, vmctx, ibs, config)
```

Comment:
I think for OpenEthereum trace, you need to somehow use `callTracer` instead of default tracer